src/MongrationDotNet/MigrationRunner.cs

```diff
@@ -29,33 +31,54 @@ public async Task Migrate()
                 logger?.LogInformation(LoggingEvents.MigrationStarted,
                     "Migration started for type: {type}, version: {version} and description: {description} ",
                     migrationCollection.Type, migrationCollection.Version.ToString(), migrationCollection.Description);
-                
-                var latestAppliedMigration = await migrationDetailsCollection
+
+            var latestAppliedMigration = await migrationDetailsCollection
                     .Find(x => x.Version == migrationCollection.Version)
                     .FirstOrDefaultAsync();
 
-                if (latestAppliedMigration != null)
-                {
-                    logger?.LogInformation(LoggingEvents.MigrationSkipped,
-                        "Migration has already been applied. Skipped migration for type: {type}, version: {version} and description: {description} ",
-                        migrationCollection.Type, migrationCollection.Version.ToString(),
-                        migrationCollection.Description);
-                    continue;
-                }
+            if (latestAppliedMigration != null)
+            {
+                logger?.LogInformation(LoggingEvents.MigrationSkipped,
+                    "Migration has already been applied or in progress. Skipped migration for type: {type}, version: {version} and description: {description} ",
+                    migrationCollection.Type, migrationCollection.Version.ToString(),
+                    migrationCollection.Description);
+                continue;
```

Comment:
You have a race condition - you need to wait until an in progress migration is complete:

Node 1 starts, there are no migrations yet so starts Migration 1.
Node 2 starts, Migration 1 is already in progress so it starts Migration 2.
 - this is bad, node 2 should wait until Migration 1 is completed (by polling DB) before trying to start Migration 2